Dislocation Motion.nlogo

```diff
@@ -156,38 +149,55 @@ to setup-atoms-and-links-and-fls
       create-fl-link-with one-of other fl-ends with [xcor = left-fl]]
     ask fl-ends [
       set color white
-      ifelse force-mode = "Tension" [ set heading 270 ] [set heading 90 ]
+      set heading 270
     ]
     if force-mode = "Tension" [
       set prev-length orig-length
-      set f-app-auto f-app
     ]
   ]
-  force-mode = "Shear" [
-    create-fl-ends 2
-    set upper-left-fl min [xcor] of atoms with [ ycor >= median-ycor ]
-    ask one-of fl-ends with [xcor = 0 and ycor = 0] [
-      set xcor upper-left-fl
-      set ycor ymax + 2 ]
-    ask one-of fl-ends with [xcor = 0 and ycor = 0] [
-      set xcor upper-left-fl
-      set ycor median-ycor
-      hide-turtle
-      create-fl-link-with one-of other fl-ends]
-    ask fl-ends [
-      set color white
-      set heading 90 ]
+    force-mode = "Compression" [
+      create-fl-ends 2
+      set left-fl xmin
+      set right-edge xmax
+      set orig-length right-edge - left-fl
+      ask one-of fl-ends with [xcor = 0 and ycor = 0] [
+        set xcor left-fl
+        set ycor max-pycor - 2 ]
+      ask one-of fl-ends with [xcor = 0 and ycor = 0] [
+        set xcor left-fl
+        set ycor min-pycor + 2
+        create-fl-link-with one-of other fl-ends with [xcor = left-fl]]
+      ask fl-ends [
+        set color white
+        set heading 90
+      ]
+    ]
+    force-mode = "Shear" [
+      create-fl-ends 2
+      set upper-left-fl min [xcor] of atoms with [ ycor >= median-ycor ]
+      ask one-of fl-ends with [xcor = 0 and ycor = 0] [
+        set xcor upper-left-fl
+        set ycor ymax + 2 ]
+      ask one-of fl-ends with [xcor = 0 and ycor = 0] [
+        set xcor upper-left-fl
+        set ycor median-ycor
+        hide-turtle
+        create-fl-link-with one-of other fl-ends]
+      ask fl-ends [
+        set color white
+        set heading 90 ]
   ])
   ask fl-links [
     set color white
   ]
-  ask atoms with [pinned? = 1] [ set shape "circle-x"]
+  ask atoms with [pinned? = True] [ set shape "circle-x"]
```

Comment:
you don't need to check `[pinned? = True]`. You can just do `[pinned?]` since pinned? itself is either True or False.